lib/new_relic/telemetry_sdk/harvester.rb

```diff
@@ -0,0 +1,66 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-telemetry-sdk-ruby/blob/main/LICENSE for complete details.
+
+module NewRelic
+  module TelemetrySdk
+    class Harvester
+
+      attr_reader :interval
+
+      def initialize interval = 5
+        @interval = interval
+        @harvestables = {}
+        @shutdown = false
+        @running = false
+        @lock = Mutex.new
+      end
+
+      def register name, buffer, client
+        @lock.synchronize do 
+          @harvestables[name] = {
+            buffer: buffer,
+            client: client
+          }
+        end
+      end
+
+      def lookup name 
+        @harvestables[name]
+      end
```

Comment:
Is this method utilized?  Seems like all the code herein references `harvestables ` directly rather than through `lookup` (except for the one test case)

Reply:
Yeah, I got the idea for it from that other nr telemetry sdk. I'm not sure if it's necessary but it was convenient to use for testing. Seems like it could be useful for other people potentially though. What do you think?

Reply:
Testing convenience is good, IMO. The other reason to provide this convenience would be if one needed to check the contents of a harvestable that had previously been registered. I'm not sure when that need would arise.

Reply:
Suggestion:  consider calling it `[]` instead of `lookup`.  Then, instead of `harvester.lookup('test')` we can have `harvester['test']` to access those harvestables.